### DISCUSSION TITLE
How to set a large amount of environment variables?

### DISCUSSION THREAD

**Discussion by WilliamBlais** [vercel/vercel#4784] [Help] [answered]
Hey,

I have ~40 variables in my .env file. Is there a way to "sync" the .env file to the environment variables? I wish that there would be a "vercel env push".

Thank you!

**Comment by rokinsky** [comment]
UP! It would be nice to have this feature:
```bash
vercel env push [environment] [file]
```

**Reply by styfle** [reply]
We might add a `vercel env push` command in the future, thanks! (internal request tracked in Story 2195)

**Reply by rokinsky** [reply]
@styfle cool, thank you a lot!

**Reply by Pytal** [reply]
@styfle any suggestions on resolving Pytal/deploy-env#16?

**Reply by styfle** [reply]
Status code [429](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) is a rate limit. You'll need to wait a bit before making more API requests.

> Too many requests - try again in 38 seconds

**Reply by Pytal** [reply]
Yeah, I was hoping maybe there was an internal non-rate limited route that could be used, although I'm assuming you'd use that for `vercel env push`

**Reply by jacksonblankenship** [reply]
This would be unbelievably helpful 🚀

**Comment by justin-elias** [comment]
For a quick solution, I wrote a shell script that will upload all env variables from a file to an environment

```#!/bin/bash
while IFS== read -r name value
do
  echo "$value" | vercel env add "$name" "$2"
done < "$1"
```

command: 
`<script name> <file> <production | preview | development>`

**Comment by Pytal** [comment, marked as answer]
Hey @WilliamBlais I made [deploy-env](https://github.com/Pytal/deploy-env) to solve this exact issue, feel free to give it a whirl and let me know how it works out for you!
